src/main/java/appeng/client/gui/me/common/MEStorageScreen.java

```diff
@@ -362,6 +362,16 @@ public boolean mouseClicked(double xCoord, double yCoord, int btn) {
             return true;
         }
 
+        // handler for middle mouse button crafting in survival mode
+        boolean isKeyPickItem = this.minecraft.options.keyPickItem.matchesMouse(btn);
+        if (isKeyPickItem) {
+            Slot slot = this.getSlot((int) xCoord, (int) yCoord);
+            if (isKeyPickItem && slot instanceof RepoSlot repoSlot && repoSlot.isCraftable()) {
```

Comment:
The `isKeyPickItem` check is done twice

```suggestion
            if (slot instanceof RepoSlot repoSlot && repoSlot.isCraftable()) {
```